
The task is to create a text classification model predicting whether an email is spam by using state-of-the-art NLP module BERT with the newest Tensorflow 1.0 API. The model again has an astonishing performance.

Though TensorFlow has moved to the 2.0 era, TensorFlow still has its popularity given its legacy and fame. Therefore, the approch provides the flexibility of running the Bert module when the corporate ML system has no short-term plan on upgrading TensorFlow to 2.0.

### Enviroment Install

In [ ]:
# Downgrade to lower version TensorFlow (TensorFlow 1.0)
!pip install tensorflow==1.15

     |████████████████████████████████| 412.3MB 40kB/s 
     |████████████████████████████████| 51kB 3.2MB/s 
     |████████████████████████████████| 3.8MB 46.9MB/s 
     |████████████████████████████████| 512kB 42.5MB/s 
  Created wheel for gast: filename=gast-0.2.2-cp36-none-any.whl size=7542 sha256=24428294fe2c491d195c1ca9bc7861153ff3da060a82a7754b8db3d3be9e5f30
  Stored in directory: /root/.cache/pip/wheels/5c/2e/7e/a1d4d4fcebe6c381f378ce7743a3ced3699feb89bcfbdadadd
Successfully built gast
ERROR: tensorflow-probability 0.11.0 has requirement gast>=0.3.2, but you'll have gast 0.2.2 which is incompatible.
  Found existing installation: tensorboard 2.3.0
    Uninstalling tensorboard-2.3.0:
      Successfully uninstalled tensorboard-2.3.0
  Found existing installation: tensorflow-estimator 2.3.0
    Uninstalling tensorflow-estimator-2.3.0:
      Successfully uninstalled tensorflow-estimator-2.3.0
  Found existing installation: gast 0.3.3
    Uninstalling gast-0.3.3:
      Successfully 

In [ ]:
from sklearn.model_selection import train_test_split
import pandas as pd
from pandas import DataFrame
import tensorflow as tf
import tensorflow_hub as hub
from datetime import datetime


In [ ]:
!pip install bert-tensorflow==1.0.1

     |████████████████████████████████| 71kB 4.1MB/s 


In [ ]:

import bert
from bert import run_classifier
from bert import optimization
from bert import tokenization

In [ ]:
!nvidia-smi

Fri Sep  4 22:45:01 2020       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 450.66       Driver Version: 418.67       CUDA Version: 10.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla V100-SXM2...  Off  | 00000000:00:04.0 Off |                    0 |
| N/A   33C    P0    24W / 300W |      0MiB / 16130MiB |      0%      Default |
|                               |                      |                 ERR! |
+-------------------------------+----------------------+----------------------+
                                                                               
+-------

In [ ]:
# Mount google drive
from google.colab import drive
drive.mount('/content/drive')

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3aietf%3awg%3aoauth%3a2.0%3aoob&scope=email%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdocs.test%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive.photos.readonly%20https%3a%2f%2fwww.googleapis.com%2fauth%2fpeopleapi.readonly&response_type=code

Enter your authorization code:
··········
Mounted at /content/drive


In [ ]:
# Download the pre-trained Bert module

BERT_MODEL_HUB = "https://tfhub.dev/google/bert_uncased_L-12_H-768_A-12/1"
bert_module = hub.Module(BERT_MODEL_HUB)

bert_module.get_signature_names()

['mlm', 'tokens', 'tokenization_info']

### Data Processing

In [ ]:
# load the saved combined file
import pickle

with open('/content/drive/My Drive/dataset.data', 'rb') as file:
    # read the data as binary data stream
    dataset = pickle.load(file)

type(dataset)

list

In [ ]:
# Though not necessary into a dataframe, doing so just for easier data manipulation when needed

Data=DataFrame(dataset, columns=['Label', 'Text'])
Data.head()

Label                                               Text
0      1  Subject: re : account # 69627 u\nhello ,\nwe s...
1      1  Subject: get your free viagra\nalways wanted t...
2      1  Subject: epson inkjet cartridges from 5 . 99\n...
3      1  Subject: again adobe photoshop ; xp pro ( sp 2...
4      1  Subject: meet ppl in your area !\n100 % free a...

Skipping data diagnoses as the work has done repeatedly in other work samples.

In [ ]:
# Split data for training and testing

train, test = train_test_split(Data, test_size = 0.20, shuffle=True, random_state=80)


In [ ]:
# Class ratios for training and testing
print('training set Spam(1):Ham(0): {:.2f}'.format(sum(train['Label'])/len(train['Label'])))
print('testing set Spam(1):Ham(0): {:.2f}'.format(sum(test['Label'])/len(test['Label'])))

training set Spam(1):Ham(0): 0.75
testing set Spam(1):Ham(0): 0.74


In [ ]:
# Use the InputExample class from BERT's run_classifier code to create examples from the data
# create objects for each of the features (guid, text_a/b, label) for all the records in our dataset 

train_InputExamples = train.apply(lambda x: bert.run_classifier.InputExample(guid=None, # Globally unique ID for bookkeeping, unused in this example
                                                                   text_a = x['Text'], 
                                                                   text_b = None, 
                                                                   label = x['Label']), axis = 1)

test_InputExamples = test.apply(lambda x: bert.run_classifier.InputExample(guid=None, 
                                                                   text_a = x['Text'], 
                                                                   text_b = None, 
                                                                   label = x['Label']), axis = 1)

In [ ]:
# Set up tokenizer

def create_tokenizer_from_hub_module():
  """Get the vocab file and casing info from the Hub module."""
  with tf.Graph().as_default():
    bert_module = hub.Module(BERT_MODEL_HUB)
    tokenization_info = bert_module(signature="tokenization_info", as_dict=True)
    with tf.Session() as sess:
      vocab_file, do_lower_case = sess.run([tokenization_info["vocab_file"],
                                            tokenization_info["do_lower_case"]])
      
  return bert.tokenization.FullTokenizer(
      vocab_file=vocab_file, do_lower_case=do_lower_case)

tokenizer = create_tokenizer_from_hub_module()

INFO:tensorflow:Saver not created because there are no variables in the graph to restore


INFO:tensorflow:Saver not created because there are no variables in the graph to restore


In [ ]:
# Create input feature
max_length = 200
label_list = [0, 1]
train_features = bert.run_classifier.convert_examples_to_features(train_InputExamples, label_list, max_length, tokenizer)
test_features = bert.run_classifier.convert_examples_to_features(test_InputExamples, label_list, max_length, tokenizer)

INFO:tensorflow:Writing example 0 of 4799


INFO:tensorflow:Writing example 0 of 4799


INFO:tensorflow:*** Example ***


INFO:tensorflow:*** Example ***


INFO:tensorflow:guid: None


INFO:tensorflow:guid: None


INFO:tensorflow:tokens: [CLS] subject : congratulations ! winning notification : verification process ( required details ) attention latin ##world promo lottery , ontario canada . from : the desk of the promotions manager , international promotions / prize award department , ref : im ##f ##d / 37 ##85 ##64 ##32 ##6 ##14 ##0 ##37 / 04 batch : 24 / 00 ##31 ##9 / is ##p . attention re / award notification ; final notice we are pleased to inform you of the announcement today of winners of the latin ##world promo lottery , ontario can ##anda international programs held on 21 st october , 2004 . your email address was attached to ticket number 123 - 01 ##48 - 79 ##0 - 45 ##9 ; with serial number 70 ##7 ##3 - 21 drew the lucky numbers 43 - 11 - 34 - 37 - 10 - 43 , and consequently won the lottery in the 2 n ##d category . you have therefore been approved for a lump sum pay out of us $ 1 , 000 . 000 . 00 in cash credited to file ref im ##f ##d / 37 ##85 ##64 ##32 ##6 ##14 ##0 [SEP]


INFO:tensorflow:tokens: [CLS] subject : congratulations ! winning notification : verification process ( required details ) attention latin ##world promo lottery , ontario canada . from : the desk of the promotions manager , international promotions / prize award department , ref : im ##f ##d / 37 ##85 ##64 ##32 ##6 ##14 ##0 ##37 / 04 batch : 24 / 00 ##31 ##9 / is ##p . attention re / award notification ; final notice we are pleased to inform you of the announcement today of winners of the latin ##world promo lottery , ontario can ##anda international programs held on 21 st october , 2004 . your email address was attached to ticket number 123 - 01 ##48 - 79 ##0 - 45 ##9 ; with serial number 70 ##7 ##3 - 21 drew the lucky numbers 43 - 11 - 34 - 37 - 10 - 43 , and consequently won the lottery in the 2 n ##d category . you have therefore been approved for a lump sum pay out of us $ 1 , 000 . 000 . 00 in cash credited to file ref im ##f ##d / 37 ##85 ##64 ##32 ##6 ##14 ##0 [SEP]


INFO:tensorflow:input_ids: 101 3395 1024 23156 999 3045 26828 1024 22616 2832 1006 3223 4751 1007 3086 3763 11108 19430 15213 1010 4561 2710 1012 2013 1024 1996 4624 1997 1996 15365 3208 1010 2248 15365 1013 3396 2400 2533 1010 25416 1024 10047 2546 2094 1013 4261 27531 21084 16703 2575 16932 2692 24434 1013 5840 14108 1024 2484 1013 4002 21486 2683 1013 2003 2361 1012 3086 2128 1013 2400 26828 1025 2345 5060 2057 2024 7537 2000 12367 2017 1997 1996 8874 2651 1997 4791 1997 1996 3763 11108 19430 15213 1010 4561 2064 13832 2248 3454 2218 2006 2538 2358 2255 1010 2432 1012 2115 10373 4769 2001 4987 2000 7281 2193 13138 1011 5890 18139 1011 6535 2692 1011 3429 2683 1025 2007 7642 2193 3963 2581 2509 1011 2538 3881 1996 5341 3616 4724 1011 2340 1011 4090 1011 4261 1011 2184 1011 4724 1010 1998 8821 2180 1996 15213 1999 1996 1016 1050 2094 4696 1012 2017 2031 3568 2042 4844 2005 1037 15116 7680 3477 2041 1997 2149 1002 1015 1010 2199 1012 2199 1012 4002 1999 5356 5827 2000 5371 25416 10047 

INFO:tensorflow:input_ids: 101 3395 1024 23156 999 3045 26828 1024 22616 2832 1006 3223 4751 1007 3086 3763 11108 19430 15213 1010 4561 2710 1012 2013 1024 1996 4624 1997 1996 15365 3208 1010 2248 15365 1013 3396 2400 2533 1010 25416 1024 10047 2546 2094 1013 4261 27531 21084 16703 2575 16932 2692 24434 1013 5840 14108 1024 2484 1013 4002 21486 2683 1013 2003 2361 1012 3086 2128 1013 2400 26828 1025 2345 5060 2057 2024 7537 2000 12367 2017 1997 1996 8874 2651 1997 4791 1997 1996 3763 11108 19430 15213 1010 4561 2064 13832 2248 3454 2218 2006 2538 2358 2255 1010 2432 1012 2115 10373 4769 2001 4987 2000 7281 2193 13138 1011 5890 18139 1011 6535 2692 1011 3429 2683 1025 2007 7642 2193 3963 2581 2509 1011 2538 3881 1996 5341 3616 4724 1011 2340 1011 4090 1011 4261 1011 2184 1011 4724 1010 1998 8821 2180 1996 15213 1999 1996 1016 1050 2094 4696 1012 2017 2031 3568 2042 4844 2005 1037 15116 7680 3477 2041 1997 2149 1002 1015 1010 2199 1012 2199 1012 4002 1999 5356 5827 2000 5371 25416 10047 

INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:label: 1 (id = 1)


INFO:tensorflow:label: 1 (id = 1)


INFO:tensorflow:*** Example ***


INFO:tensorflow:*** Example ***


INFO:tensorflow:guid: None


INFO:tensorflow:guid: None


INFO:tensorflow:tokens: [CLS] subject : ware ##z cd ' s microsoft windows xp professional 2002 $ . 50 retail pr . ice : $ . 270 . 99 retail pr . ice : $ . 50 . 00 you s . ave : $ . 220 . 00 b . u ##y it n . ow . ! microsoft windows 2000 professional $ . 50 retail pr . ice : $ . 266 . 99 retail pr . ice : $ . 50 . 00 you s . ave : $ . 216 . 99 b . u ##y it n . ow . ! microsoft office xp professional 2002 $ . 100 retail pr . ice : $ . 57 ##9 . 99 retail pr . ice : $ . 100 . 00 you s . ave : $ . 480 . 00 b . u ##y it n . ow . ! adobe page ##maker 7 . 0 $ . 60 retail pr . ice : $ . 404 . 99 retail pr . ice : $ . 60 . 00 you s . ave : $ . 344 . 00 b . u ##y it n . ow [SEP]


INFO:tensorflow:tokens: [CLS] subject : ware ##z cd ' s microsoft windows xp professional 2002 $ . 50 retail pr . ice : $ . 270 . 99 retail pr . ice : $ . 50 . 00 you s . ave : $ . 220 . 00 b . u ##y it n . ow . ! microsoft windows 2000 professional $ . 50 retail pr . ice : $ . 266 . 99 retail pr . ice : $ . 50 . 00 you s . ave : $ . 216 . 99 b . u ##y it n . ow . ! microsoft office xp professional 2002 $ . 100 retail pr . ice : $ . 57 ##9 . 99 retail pr . ice : $ . 100 . 00 you s . ave : $ . 480 . 00 b . u ##y it n . ow . ! adobe page ##maker 7 . 0 $ . 60 retail pr . ice : $ . 404 . 99 retail pr . ice : $ . 60 . 00 you s . ave : $ . 344 . 00 b . u ##y it n . ow [SEP]


INFO:tensorflow:input_ids: 101 3395 1024 16283 2480 3729 1005 1055 7513 3645 26726 2658 2526 1002 1012 2753 7027 10975 1012 3256 1024 1002 1012 13756 1012 5585 7027 10975 1012 3256 1024 1002 1012 2753 1012 4002 2017 1055 1012 13642 1024 1002 1012 10545 1012 4002 1038 1012 1057 2100 2009 1050 1012 27593 1012 999 7513 3645 2456 2658 1002 1012 2753 7027 10975 1012 3256 1024 1002 1012 25162 1012 5585 7027 10975 1012 3256 1024 1002 1012 2753 1012 4002 2017 1055 1012 13642 1024 1002 1012 20294 1012 5585 1038 1012 1057 2100 2009 1050 1012 27593 1012 999 7513 2436 26726 2658 2526 1002 1012 2531 7027 10975 1012 3256 1024 1002 1012 5401 2683 1012 5585 7027 10975 1012 3256 1024 1002 1012 2531 1012 4002 2017 1055 1012 13642 1024 1002 1012 17295 1012 4002 1038 1012 1057 2100 2009 1050 1012 27593 1012 999 18106 3931 8571 1021 1012 1014 1002 1012 3438 7027 10975 1012 3256 1024 1002 1012 24837 1012 5585 7027 10975 1012 3256 1024 1002 1012 3438 1012 4002 2017 1055 1012 13642 1024 1002 1012 29386 1012 4

INFO:tensorflow:input_ids: 101 3395 1024 16283 2480 3729 1005 1055 7513 3645 26726 2658 2526 1002 1012 2753 7027 10975 1012 3256 1024 1002 1012 13756 1012 5585 7027 10975 1012 3256 1024 1002 1012 2753 1012 4002 2017 1055 1012 13642 1024 1002 1012 10545 1012 4002 1038 1012 1057 2100 2009 1050 1012 27593 1012 999 7513 3645 2456 2658 1002 1012 2753 7027 10975 1012 3256 1024 1002 1012 25162 1012 5585 7027 10975 1012 3256 1024 1002 1012 2753 1012 4002 2017 1055 1012 13642 1024 1002 1012 20294 1012 5585 1038 1012 1057 2100 2009 1050 1012 27593 1012 999 7513 2436 26726 2658 2526 1002 1012 2531 7027 10975 1012 3256 1024 1002 1012 5401 2683 1012 5585 7027 10975 1012 3256 1024 1002 1012 2531 1012 4002 2017 1055 1012 13642 1024 1002 1012 17295 1012 4002 1038 1012 1057 2100 2009 1050 1012 27593 1012 999 18106 3931 8571 1021 1012 1014 1002 1012 3438 7027 10975 1012 3256 1024 1002 1012 24837 1012 5585 7027 10975 1012 3256 1024 1002 1012 3438 1012 4002 2017 1055 1012 13642 1024 1002 1012 29386 1012 4

INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:label: 1 (id = 1)


INFO:tensorflow:label: 1 (id = 1)


INFO:tensorflow:*** Example ***


INFO:tensorflow:*** Example ***


INFO:tensorflow:guid: None


INFO:tensorflow:guid: None


INFO:tensorflow:tokens: [CLS] subject : is this helpful for you ? chart ##reus ##e we carry wholesale rep ##li ##ica bags , watches or whatever that you will be proud to own ! we are the actual site who knows what we are doing , quoting a fair retail price and offering personal contact if preferred . book ##mark us at http : / / y ##y . moons ##park ##leg ##rea ##t . com / li / add ##ida ##s , ball ##y , b ##v ##lga ##ri , bu ##rber ##ry , cart ##ier , chan ##el , christian di ##or , dun ##hill , dupont , es ##ca ##da , fen ##di , fe ##rra ##gam ##o , gu ##cci , hermes , i ##wc , jacob & co . , louis vu ##itt ##on , mont blanc , mo ##va ##do , nike , omega , or ##is , pr ##ada , pu ##ma , ra ##do , roger dub ##uis , role ##x , sector , tag he ##uer , techno ##marine , tiffany , timber ##land , tudor who ##olo ##vin ' your smile ##lov ##in ' your styled ##on ' t you get it [SEP]


INFO:tensorflow:tokens: [CLS] subject : is this helpful for you ? chart ##reus ##e we carry wholesale rep ##li ##ica bags , watches or whatever that you will be proud to own ! we are the actual site who knows what we are doing , quoting a fair retail price and offering personal contact if preferred . book ##mark us at http : / / y ##y . moons ##park ##leg ##rea ##t . com / li / add ##ida ##s , ball ##y , b ##v ##lga ##ri , bu ##rber ##ry , cart ##ier , chan ##el , christian di ##or , dun ##hill , dupont , es ##ca ##da , fen ##di , fe ##rra ##gam ##o , gu ##cci , hermes , i ##wc , jacob & co . , louis vu ##itt ##on , mont blanc , mo ##va ##do , nike , omega , or ##is , pr ##ada , pu ##ma , ra ##do , roger dub ##uis , role ##x , sector , tag he ##uer , techno ##marine , tiffany , timber ##land , tudor who ##olo ##vin ' your smile ##lov ##in ' your styled ##on ' t you get it [SEP]


INFO:tensorflow:input_ids: 101 3395 1024 2003 2023 14044 2005 2017 1029 3673 23446 2063 2057 4287 17264 16360 3669 5555 8641 1010 12197 2030 3649 2008 2017 2097 2022 7098 2000 2219 999 2057 2024 1996 5025 2609 2040 4282 2054 2057 2024 2725 1010 27394 1037 4189 7027 3976 1998 5378 3167 3967 2065 6871 1012 2338 10665 2149 2012 8299 1024 1013 1013 1061 2100 1012 23377 14432 23115 16416 2102 1012 4012 1013 5622 1013 5587 8524 2015 1010 3608 2100 1010 1038 2615 27887 3089 1010 20934 20473 2854 1010 11122 3771 1010 9212 2884 1010 3017 4487 2953 1010 24654 7100 1010 22848 1010 9686 3540 2850 1010 21713 4305 1010 10768 11335 22864 2080 1010 19739 14693 1010 24127 1010 1045 16526 1010 6213 1004 2522 1012 1010 3434 24728 12474 2239 1010 18318 18698 1010 9587 3567 3527 1010 18368 1010 14827 1010 2030 2483 1010 10975 8447 1010 16405 2863 1010 10958 3527 1010 5074 12931 27020 1010 2535 2595 1010 4753 1010 6415 2002 13094 1010 21416 21966 1010 14381 1010 7227 3122 1010 15588 2040 12898 6371 1005 211

INFO:tensorflow:input_ids: 101 3395 1024 2003 2023 14044 2005 2017 1029 3673 23446 2063 2057 4287 17264 16360 3669 5555 8641 1010 12197 2030 3649 2008 2017 2097 2022 7098 2000 2219 999 2057 2024 1996 5025 2609 2040 4282 2054 2057 2024 2725 1010 27394 1037 4189 7027 3976 1998 5378 3167 3967 2065 6871 1012 2338 10665 2149 2012 8299 1024 1013 1013 1061 2100 1012 23377 14432 23115 16416 2102 1012 4012 1013 5622 1013 5587 8524 2015 1010 3608 2100 1010 1038 2615 27887 3089 1010 20934 20473 2854 1010 11122 3771 1010 9212 2884 1010 3017 4487 2953 1010 24654 7100 1010 22848 1010 9686 3540 2850 1010 21713 4305 1010 10768 11335 22864 2080 1010 19739 14693 1010 24127 1010 1045 16526 1010 6213 1004 2522 1012 1010 3434 24728 12474 2239 1010 18318 18698 1010 9587 3567 3527 1010 18368 1010 14827 1010 2030 2483 1010 10975 8447 1010 16405 2863 1010 10958 3527 1010 5074 12931 27020 1010 2535 2595 1010 4753 1010 6415 2002 13094 1010 21416 21966 1010 14381 1010 7227 3122 1010 15588 2040 12898 6371 1005 211

INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:label: 1 (id = 1)


INFO:tensorflow:label: 1 (id = 1)


INFO:tensorflow:*** Example ***


INFO:tensorflow:*** Example ***


INFO:tensorflow:guid: None


INFO:tensorflow:guid: None


INFO:tensorflow:tokens: [CLS] subject : low interest rate on mor [ gage loans hi i have tried contact ##ing you before in regards to the promised low interest mort ##ga ( ge rate . the good news is that you did qualify for the loan , for a rate of 4 . 2 % ! you could receive over u 380 , 000 for as little as i 500 per month . ba ( d credit ? doesn ' t matter , low rates are fixed no matter what ! to get a fr ! ee , no ob ##li , ga ##tion consultation click below : have a great day ! lima moore l ##h ##zh ##k h ##w ##m ##f ##v ##t [SEP]


INFO:tensorflow:tokens: [CLS] subject : low interest rate on mor [ gage loans hi i have tried contact ##ing you before in regards to the promised low interest mort ##ga ( ge rate . the good news is that you did qualify for the loan , for a rate of 4 . 2 % ! you could receive over u 380 , 000 for as little as i 500 per month . ba ( d credit ? doesn ' t matter , low rates are fixed no matter what ! to get a fr ! ee , no ob ##li , ga ##tion consultation click below : have a great day ! lima moore l ##h ##zh ##k h ##w ##m ##f ##v ##t [SEP]


INFO:tensorflow:input_ids: 101 3395 1024 2659 3037 3446 2006 22822 1031 10012 10940 7632 1045 2031 2699 3967 2075 2017 2077 1999 12362 2000 1996 5763 2659 3037 22294 3654 1006 16216 3446 1012 1996 2204 2739 2003 2008 2017 2106 7515 2005 1996 5414 1010 2005 1037 3446 1997 1018 1012 1016 1003 999 2017 2071 4374 2058 1057 17014 1010 2199 2005 2004 2210 2004 1045 3156 2566 3204 1012 8670 1006 1040 4923 1029 2987 1005 1056 3043 1010 2659 6165 2024 4964 2053 3043 2054 999 2000 2131 1037 10424 999 25212 1010 2053 27885 3669 1010 11721 3508 16053 11562 2917 1024 2031 1037 2307 2154 999 12967 5405 1048 2232 27922 2243 1044 2860 2213 2546 2615 2102 102 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:input_ids: 101 3395 1024 2659 3037 3446 2006 22822 1031 10012 10940 7632 1045 2031 2699 3967 2075 2017 2077 1999 12362 2000 1996 5763 2659 3037 22294 3654 1006 16216 3446 1012 1996 2204 2739 2003 2008 2017 2106 7515 2005 1996 5414 1010 2005 1037 3446 1997 1018 1012 1016 1003 999 2017 2071 4374 2058 1057 17014 1010 2199 2005 2004 2210 2004 1045 3156 2566 3204 1012 8670 1006 1040 4923 1029 2987 1005 1056 3043 1010 2659 6165 2024 4964 2053 3043 2054 999 2000 2131 1037 10424 999 25212 1010 2053 27885 3669 1010 11721 3508 16053 11562 2917 1024 2031 1037 2307 2154 999 12967 5405 1048 2232 27922 2243 1044 2860 2213 2546 2615 2102 102 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:label: 1 (id = 1)


INFO:tensorflow:label: 1 (id = 1)


INFO:tensorflow:*** Example ***


INFO:tensorflow:*** Example ***


INFO:tensorflow:guid: None


INFO:tensorflow:guid: None


INFO:tensorflow:tokens: [CLS] subject : re : trans ##west ##ern weekly report kern river ' s through ##put to ca was only 59 % ? did they have an out ##age ? is the remaining gas dropping - off before ca ? we usually assume in our gas flows analysis that kern is 100 % full . this seems unusual . christine stokes 09 / 15 / 2000 08 : 46 am to : steven harris / et & s / en ##ron @ en ##ron , kevin h ##yat ##t / et & s / en ##ron @ en ##ron , t ##k lo ##hman / et & s / en ##ron @ en ##ron , jeff ##ery fa ##wc ##ett / et & s / en ##ron @ en ##ron , lorraine lin ##dberg / et & s / en ##ron @ en ##ron , michelle lok ##ay / et & s / en ##ron @ en ##ron , lin ##dy don ##oh ##o / et & s / en ##ron @ en ##ron cc : subject : trans ##west ##ern weekly report for the week september 11 - 15 th : trans ##west ##ern weekly average deliveries to [SEP]


INFO:tensorflow:tokens: [CLS] subject : re : trans ##west ##ern weekly report kern river ' s through ##put to ca was only 59 % ? did they have an out ##age ? is the remaining gas dropping - off before ca ? we usually assume in our gas flows analysis that kern is 100 % full . this seems unusual . christine stokes 09 / 15 / 2000 08 : 46 am to : steven harris / et & s / en ##ron @ en ##ron , kevin h ##yat ##t / et & s / en ##ron @ en ##ron , t ##k lo ##hman / et & s / en ##ron @ en ##ron , jeff ##ery fa ##wc ##ett / et & s / en ##ron @ en ##ron , lorraine lin ##dberg / et & s / en ##ron @ en ##ron , michelle lok ##ay / et & s / en ##ron @ en ##ron , lin ##dy don ##oh ##o / et & s / en ##ron @ en ##ron cc : subject : trans ##west ##ern weekly report for the week september 11 - 15 th : trans ##west ##ern weekly average deliveries to [SEP]


INFO:tensorflow:input_ids: 101 3395 1024 2128 1024 9099 19650 11795 4882 3189 22762 2314 1005 1055 2083 18780 2000 6187 2001 2069 5354 1003 1029 2106 2027 2031 2019 2041 4270 1029 2003 1996 3588 3806 7510 1011 2125 2077 6187 1029 2057 2788 7868 1999 2256 3806 6223 4106 2008 22762 2003 2531 1003 2440 1012 2023 3849 5866 1012 10941 18716 5641 1013 2321 1013 2456 5511 1024 4805 2572 2000 1024 7112 5671 1013 3802 1004 1055 1013 4372 4948 1030 4372 4948 1010 4901 1044 26139 2102 1013 3802 1004 1055 1013 4372 4948 1030 4372 4948 1010 1056 2243 8840 13890 1013 3802 1004 1055 1013 4372 4948 1030 4372 4948 1010 5076 7301 6904 16526 6582 1013 3802 1004 1055 1013 4372 4948 1030 4372 4948 1010 13895 11409 25190 1013 3802 1004 1055 1013 4372 4948 1030 4372 4948 1010 9393 13660 4710 1013 3802 1004 1055 1013 4372 4948 1030 4372 4948 1010 11409 5149 2123 11631 2080 1013 3802 1004 1055 1013 4372 4948 1030 4372 4948 10507 1024 3395 1024 9099 19650 11795 4882 3189 2005 1996 2733 2244 2340 1011 2321 16215

INFO:tensorflow:input_ids: 101 3395 1024 2128 1024 9099 19650 11795 4882 3189 22762 2314 1005 1055 2083 18780 2000 6187 2001 2069 5354 1003 1029 2106 2027 2031 2019 2041 4270 1029 2003 1996 3588 3806 7510 1011 2125 2077 6187 1029 2057 2788 7868 1999 2256 3806 6223 4106 2008 22762 2003 2531 1003 2440 1012 2023 3849 5866 1012 10941 18716 5641 1013 2321 1013 2456 5511 1024 4805 2572 2000 1024 7112 5671 1013 3802 1004 1055 1013 4372 4948 1030 4372 4948 1010 4901 1044 26139 2102 1013 3802 1004 1055 1013 4372 4948 1030 4372 4948 1010 1056 2243 8840 13890 1013 3802 1004 1055 1013 4372 4948 1030 4372 4948 1010 5076 7301 6904 16526 6582 1013 3802 1004 1055 1013 4372 4948 1030 4372 4948 1010 13895 11409 25190 1013 3802 1004 1055 1013 4372 4948 1030 4372 4948 1010 9393 13660 4710 1013 3802 1004 1055 1013 4372 4948 1030 4372 4948 1010 11409 5149 2123 11631 2080 1013 3802 1004 1055 1013 4372 4948 1030 4372 4948 10507 1024 3395 1024 9099 19650 11795 4882 3189 2005 1996 2733 2244 2340 1011 2321 16215

INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:label: 0 (id = 0)


INFO:tensorflow:label: 0 (id = 0)


INFO:tensorflow:Writing example 0 of 1200


INFO:tensorflow:Writing example 0 of 1200


INFO:tensorflow:*** Example ***


INFO:tensorflow:*** Example ***


INFO:tensorflow:guid: None


INFO:tensorflow:guid: None


INFO:tensorflow:tokens: [CLS] subject : f ##w : article - - - - - original message - - - - - from : taylor , gina sent : friday , february 15 , 2002 4 : 20 pm to : harris , steven ; h ##yat ##t , kevin subject : article i wanted to be sure you saw this . gina development of sonora ##n pipeline project halted 02 / 15 / 2002 pr news ##wire ( copyright ( c ) 2002 , pr news ##wire ) houston , feb . 15 / pr ##ne ##ws ##wire - first ##cal ##l / - - kind ##er morgan energy partners , l . p . ( ny ##se : km ##p ) and cal ##pine corporation ( ny ##se : cp ##n ) today announced the termination of the proposed sonora ##n pipeline project . the sonora ##n pipeline was unable to secure sufficient binding commitments to make a successful project given market conditions . securing firm transportation agreements was a st ##ip ##ulation that executives of both companies initially stated was necessary before development of the project could begin . neither company has spent significant dollars on the pr

INFO:tensorflow:tokens: [CLS] subject : f ##w : article - - - - - original message - - - - - from : taylor , gina sent : friday , february 15 , 2002 4 : 20 pm to : harris , steven ; h ##yat ##t , kevin subject : article i wanted to be sure you saw this . gina development of sonora ##n pipeline project halted 02 / 15 / 2002 pr news ##wire ( copyright ( c ) 2002 , pr news ##wire ) houston , feb . 15 / pr ##ne ##ws ##wire - first ##cal ##l / - - kind ##er morgan energy partners , l . p . ( ny ##se : km ##p ) and cal ##pine corporation ( ny ##se : cp ##n ) today announced the termination of the proposed sonora ##n pipeline project . the sonora ##n pipeline was unable to secure sufficient binding commitments to make a successful project given market conditions . securing firm transportation agreements was a st ##ip ##ulation that executives of both companies initially stated was necessary before development of the project could begin . neither company has spent significant dollars on the pr

INFO:tensorflow:input_ids: 101 3395 1024 1042 2860 1024 3720 1011 1011 1011 1011 1011 2434 4471 1011 1011 1011 1011 1011 2013 1024 4202 1010 17508 2741 1024 5958 1010 2337 2321 1010 2526 1018 1024 2322 7610 2000 1024 5671 1010 7112 1025 1044 26139 2102 1010 4901 3395 1024 3720 1045 2359 2000 2022 2469 2017 2387 2023 1012 17508 2458 1997 26647 2078 13117 2622 12705 6185 1013 2321 1013 2526 10975 2739 20357 1006 9385 1006 1039 1007 2526 1010 10975 2739 20357 1007 5395 1010 13114 1012 2321 1013 10975 2638 9333 20357 1011 2034 9289 2140 1013 1011 1011 2785 2121 5253 2943 5826 1010 1048 1012 1052 1012 1006 6396 3366 1024 2463 2361 1007 1998 10250 19265 3840 1006 6396 3366 1024 18133 2078 1007 2651 2623 1996 18287 1997 1996 3818 26647 2078 13117 2622 1012 1996 26647 2078 13117 2001 4039 2000 5851 7182 8031 17786 2000 2191 1037 3144 2622 2445 3006 3785 1012 12329 3813 5193 10540 2001 1037 2358 11514 9513 2008 12706 1997 2119 3316 3322 3090 2001 4072 2077 2458 1997 1996 2622 2071 4088 1012 444

INFO:tensorflow:input_ids: 101 3395 1024 1042 2860 1024 3720 1011 1011 1011 1011 1011 2434 4471 1011 1011 1011 1011 1011 2013 1024 4202 1010 17508 2741 1024 5958 1010 2337 2321 1010 2526 1018 1024 2322 7610 2000 1024 5671 1010 7112 1025 1044 26139 2102 1010 4901 3395 1024 3720 1045 2359 2000 2022 2469 2017 2387 2023 1012 17508 2458 1997 26647 2078 13117 2622 12705 6185 1013 2321 1013 2526 10975 2739 20357 1006 9385 1006 1039 1007 2526 1010 10975 2739 20357 1007 5395 1010 13114 1012 2321 1013 10975 2638 9333 20357 1011 2034 9289 2140 1013 1011 1011 2785 2121 5253 2943 5826 1010 1048 1012 1052 1012 1006 6396 3366 1024 2463 2361 1007 1998 10250 19265 3840 1006 6396 3366 1024 18133 2078 1007 2651 2623 1996 18287 1997 1996 3818 26647 2078 13117 2622 1012 1996 26647 2078 13117 2001 4039 2000 5851 7182 8031 17786 2000 2191 1037 3144 2622 2445 3006 3785 1012 12329 3813 5193 10540 2001 1037 2358 11514 9513 2008 12706 1997 2119 3316 3322 3090 2001 4072 2077 2458 1997 1996 2622 2071 4088 1012 444

INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:label: 0 (id = 0)


INFO:tensorflow:label: 0 (id = 0)


INFO:tensorflow:*** Example ***


INFO:tensorflow:*** Example ***


INFO:tensorflow:guid: None


INFO:tensorflow:guid: None


INFO:tensorflow:tokens: [CLS] subject : po ##go check ##list - revised contact john ha ##vard at 71 ##3 - 297 - 501 ##1 - - - - - - - - - - - - - - - - - - - - - - forward ##ed by kevin h ##yat ##t / et & s / en ##ron on 12 / 20 / 2000 02 : 01 pm - - - - - - - - - - - - - - - - - - - - - - - - - - - christine stokes 12 / 04 / 2000 03 : 13 pm to : perry frazier / et & s / en ##ron @ en ##ron cc : steve well ##er / et & s / en ##ron @ en ##ron , kevin h ##yat ##t / et & s / en ##ron @ en ##ron subject : po ##go check ##list - revised del ##ete prior check - list , the location was er ##rone ##ous . perry - attached is the check ##list and project summary for a potential po ##go well connect to trans ##west ##ern in lea county . po ##go has [SEP]


INFO:tensorflow:tokens: [CLS] subject : po ##go check ##list - revised contact john ha ##vard at 71 ##3 - 297 - 501 ##1 - - - - - - - - - - - - - - - - - - - - - - forward ##ed by kevin h ##yat ##t / et & s / en ##ron on 12 / 20 / 2000 02 : 01 pm - - - - - - - - - - - - - - - - - - - - - - - - - - - christine stokes 12 / 04 / 2000 03 : 13 pm to : perry frazier / et & s / en ##ron @ en ##ron cc : steve well ##er / et & s / en ##ron @ en ##ron , kevin h ##yat ##t / et & s / en ##ron @ en ##ron subject : po ##go check ##list - revised del ##ete prior check - list , the location was er ##rone ##ous . perry - attached is the check ##list and project summary for a potential po ##go well connect to trans ##west ##ern in lea county . po ##go has [SEP]


INFO:tensorflow:input_ids: 101 3395 1024 13433 3995 4638 9863 1011 8001 3967 2198 5292 25911 2012 6390 2509 1011 27502 1011 16202 2487 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 2830 2098 2011 4901 1044 26139 2102 1013 3802 1004 1055 1013 4372 4948 2006 2260 1013 2322 1013 2456 6185 1024 5890 7610 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 10941 18716 2260 1013 5840 1013 2456 6021 1024 2410 7610 2000 1024 6890 26551 1013 3802 1004 1055 1013 4372 4948 1030 4372 4948 10507 1024 3889 2092 2121 1013 3802 1004 1055 1013 4372 4948 1030 4372 4948 1010 4901 1044 26139 2102 1013 3802 1004 1055 1013 4372 4948 1030 4372 4948 3395 1024 13433 3995 4638 9863 1011 8001 3972 12870 3188 4638 1011 2862 1010 1996 3295 2001 9413 20793 3560 1012 6890 1011 4987 2003 1996 4638 9863 1998 2622 12654 2005 1037 4022 13433 3995 2092 7532 2000 9099 19650 117

INFO:tensorflow:input_ids: 101 3395 1024 13433 3995 4638 9863 1011 8001 3967 2198 5292 25911 2012 6390 2509 1011 27502 1011 16202 2487 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 2830 2098 2011 4901 1044 26139 2102 1013 3802 1004 1055 1013 4372 4948 2006 2260 1013 2322 1013 2456 6185 1024 5890 7610 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 1011 10941 18716 2260 1013 5840 1013 2456 6021 1024 2410 7610 2000 1024 6890 26551 1013 3802 1004 1055 1013 4372 4948 1030 4372 4948 10507 1024 3889 2092 2121 1013 3802 1004 1055 1013 4372 4948 1030 4372 4948 1010 4901 1044 26139 2102 1013 3802 1004 1055 1013 4372 4948 1030 4372 4948 3395 1024 13433 3995 4638 9863 1011 8001 3972 12870 3188 4638 1011 2862 1010 1996 3295 2001 9413 20793 3560 1012 6890 1011 4987 2003 1996 4638 9863 1998 2622 12654 2005 1037 4022 13433 3995 2092 7532 2000 9099 19650 117

INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:label: 0 (id = 0)


INFO:tensorflow:label: 0 (id = 0)


INFO:tensorflow:*** Example ***


INFO:tensorflow:*** Example ***


INFO:tensorflow:guid: None


INFO:tensorflow:guid: None


INFO:tensorflow:tokens: [CLS] subject : much better than via ##gra hello , try this revolutionary product , cia ##lis soft tab ##s . cia ##lis soft tab ##s is the new imp ##ote ##nce treatment drug that everyone is talking about . soft tab ##s acts up to 36 hours , compare this to only two or three hours of via ##gra action ! the active ingredient is tad ##ala ##fi ##l , same as in brand cia ##lis . simply dissolve half a pill under your tongue , 10 min before sex , for the best erection ##s you ' ve ever had ! soft tab ##s also have less side ##backs ( you can drive or mix alcohol drinks with them ) . no prior prescription needed . you can get it at : http : / / mon ##eti ##sed . net / soft / world r ##x direct can bring you quality generic drugs for a fraction of the cost of the expensive brand name equivalent ##s . order our tad ##ala ##fi ##l pills today and save 80 % . we ship worldwide , and currently supply to over 1 million customers globally ! we always [SEP]


INFO:tensorflow:tokens: [CLS] subject : much better than via ##gra hello , try this revolutionary product , cia ##lis soft tab ##s . cia ##lis soft tab ##s is the new imp ##ote ##nce treatment drug that everyone is talking about . soft tab ##s acts up to 36 hours , compare this to only two or three hours of via ##gra action ! the active ingredient is tad ##ala ##fi ##l , same as in brand cia ##lis . simply dissolve half a pill under your tongue , 10 min before sex , for the best erection ##s you ' ve ever had ! soft tab ##s also have less side ##backs ( you can drive or mix alcohol drinks with them ) . no prior prescription needed . you can get it at : http : / / mon ##eti ##sed . net / soft / world r ##x direct can bring you quality generic drugs for a fraction of the cost of the expensive brand name equivalent ##s . order our tad ##ala ##fi ##l pills today and save 80 % . we ship worldwide , and currently supply to over 1 million customers globally ! we always [SEP]


INFO:tensorflow:input_ids: 101 3395 1024 2172 2488 2084 3081 17643 7592 1010 3046 2023 6208 4031 1010 9915 6856 3730 21628 2015 1012 9915 6856 3730 21628 2015 2003 1996 2047 17727 12184 5897 3949 4319 2008 3071 2003 3331 2055 1012 3730 21628 2015 4490 2039 2000 4029 2847 1010 12826 2023 2000 2069 2048 2030 2093 2847 1997 3081 17643 2895 999 1996 3161 21774 2003 18819 7911 8873 2140 1010 2168 2004 1999 4435 9915 6856 1012 3432 21969 2431 1037 17357 2104 2115 4416 1010 2184 8117 2077 3348 1010 2005 1996 2190 13760 2015 2017 1005 2310 2412 2018 999 3730 21628 2015 2036 2031 2625 2217 12221 1006 2017 2064 3298 2030 4666 6544 8974 2007 2068 1007 1012 2053 3188 20422 2734 1012 2017 2064 2131 2009 2012 1024 8299 1024 1013 1013 12256 20624 6924 1012 5658 1013 3730 1013 2088 1054 2595 3622 2064 3288 2017 3737 12391 5850 2005 1037 12884 1997 1996 3465 1997 1996 6450 4435 2171 5662 2015 1012 2344 2256 18819 7911 8873 2140 15345 2651 1998 3828 3770 1003 1012 2057 2911 4969 1010 1998 2747 4425 2000

INFO:tensorflow:input_ids: 101 3395 1024 2172 2488 2084 3081 17643 7592 1010 3046 2023 6208 4031 1010 9915 6856 3730 21628 2015 1012 9915 6856 3730 21628 2015 2003 1996 2047 17727 12184 5897 3949 4319 2008 3071 2003 3331 2055 1012 3730 21628 2015 4490 2039 2000 4029 2847 1010 12826 2023 2000 2069 2048 2030 2093 2847 1997 3081 17643 2895 999 1996 3161 21774 2003 18819 7911 8873 2140 1010 2168 2004 1999 4435 9915 6856 1012 3432 21969 2431 1037 17357 2104 2115 4416 1010 2184 8117 2077 3348 1010 2005 1996 2190 13760 2015 2017 1005 2310 2412 2018 999 3730 21628 2015 2036 2031 2625 2217 12221 1006 2017 2064 3298 2030 4666 6544 8974 2007 2068 1007 1012 2053 3188 20422 2734 1012 2017 2064 2131 2009 2012 1024 8299 1024 1013 1013 12256 20624 6924 1012 5658 1013 3730 1013 2088 1054 2595 3622 2064 3288 2017 3737 12391 5850 2005 1037 12884 1997 1996 3465 1997 1996 6450 4435 2171 5662 2015 1012 2344 2256 18819 7911 8873 2140 15345 2651 1998 3828 3770 1003 1012 2057 2911 4969 1010 1998 2747 4425 2000

INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:label: 1 (id = 1)


INFO:tensorflow:label: 1 (id = 1)


INFO:tensorflow:*** Example ***


INFO:tensorflow:*** Example ***


INFO:tensorflow:guid: None


INFO:tensorflow:guid: None


INFO:tensorflow:tokens: [CLS] subject : f ##w : california ' s desperate attempts to avoid a rate increase - ce ##ra alert f ##yi - - - - - original message - - - - - from : web ##master @ ce ##ra . com @ en ##ron sent : monday , february 05 , 2001 4 : 38 pm to : insights @ ce ##ra . com subject : california ' s desperate attempts to avoid a rate increase - ce ##ra alert title : california ' s desperate attempts to avoid a rate increase ur ##l : http : / / www 20 . ce ##ra . com / ep ##ro ##fi ##le ? u = 35 & m = 224 ##1 california , s desperate attempts to avoid a rate increase on february 1 , 2001 , the california assembly approved and the california governor signed a new law intended to secure power supplies for california . the legislature has been meeting since early january in a & special extraordinary session 8 to address the power market crisis in the state , and the new law represents their first significant piece of legislation . the bill , ab [SEP]


INFO:tensorflow:tokens: [CLS] subject : f ##w : california ' s desperate attempts to avoid a rate increase - ce ##ra alert f ##yi - - - - - original message - - - - - from : web ##master @ ce ##ra . com @ en ##ron sent : monday , february 05 , 2001 4 : 38 pm to : insights @ ce ##ra . com subject : california ' s desperate attempts to avoid a rate increase - ce ##ra alert title : california ' s desperate attempts to avoid a rate increase ur ##l : http : / / www 20 . ce ##ra . com / ep ##ro ##fi ##le ? u = 35 & m = 224 ##1 california , s desperate attempts to avoid a rate increase on february 1 , 2001 , the california assembly approved and the california governor signed a new law intended to secure power supplies for california . the legislature has been meeting since early january in a & special extraordinary session 8 to address the power market crisis in the state , and the new law represents their first significant piece of legislation . the bill , ab [SEP]


INFO:tensorflow:input_ids: 101 3395 1024 1042 2860 1024 2662 1005 1055 7143 4740 2000 4468 1037 3446 3623 1011 8292 2527 9499 1042 10139 1011 1011 1011 1011 1011 2434 4471 1011 1011 1011 1011 1011 2013 1024 4773 8706 1030 8292 2527 1012 4012 1030 4372 4948 2741 1024 6928 1010 2337 5709 1010 2541 1018 1024 4229 7610 2000 1024 20062 1030 8292 2527 1012 4012 3395 1024 2662 1005 1055 7143 4740 2000 4468 1037 3446 3623 1011 8292 2527 9499 2516 1024 2662 1005 1055 7143 4740 2000 4468 1037 3446 3623 24471 2140 1024 8299 1024 1013 1013 7479 2322 1012 8292 2527 1012 4012 1013 4958 3217 8873 2571 1029 1057 1027 3486 1004 1049 1027 19711 2487 2662 1010 1055 7143 4740 2000 4468 1037 3446 3623 2006 2337 1015 1010 2541 1010 1996 2662 3320 4844 1998 1996 2662 3099 2772 1037 2047 2375 3832 2000 5851 2373 6067 2005 2662 1012 1996 6372 2038 2042 3116 2144 2220 2254 1999 1037 1004 2569 9313 5219 1022 2000 4769 1996 2373 3006 5325 1999 1996 2110 1010 1998 1996 2047 2375 5836 2037 2034 3278 3538 1997 6094 

INFO:tensorflow:input_ids: 101 3395 1024 1042 2860 1024 2662 1005 1055 7143 4740 2000 4468 1037 3446 3623 1011 8292 2527 9499 1042 10139 1011 1011 1011 1011 1011 2434 4471 1011 1011 1011 1011 1011 2013 1024 4773 8706 1030 8292 2527 1012 4012 1030 4372 4948 2741 1024 6928 1010 2337 5709 1010 2541 1018 1024 4229 7610 2000 1024 20062 1030 8292 2527 1012 4012 3395 1024 2662 1005 1055 7143 4740 2000 4468 1037 3446 3623 1011 8292 2527 9499 2516 1024 2662 1005 1055 7143 4740 2000 4468 1037 3446 3623 24471 2140 1024 8299 1024 1013 1013 7479 2322 1012 8292 2527 1012 4012 1013 4958 3217 8873 2571 1029 1057 1027 3486 1004 1049 1027 19711 2487 2662 1010 1055 7143 4740 2000 4468 1037 3446 3623 2006 2337 1015 1010 2541 1010 1996 2662 3320 4844 1998 1996 2662 3099 2772 1037 2047 2375 3832 2000 5851 2373 6067 2005 2662 1012 1996 6372 2038 2042 3116 2144 2220 2254 1999 1037 1004 2569 9313 5219 1022 2000 4769 1996 2373 3006 5325 1999 1996 2110 1010 1998 1996 2047 2375 5836 2037 2034 3278 3538 1997 6094 

INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:label: 0 (id = 0)


INFO:tensorflow:label: 0 (id = 0)


INFO:tensorflow:*** Example ***


INFO:tensorflow:*** Example ***


INFO:tensorflow:guid: None


INFO:tensorflow:guid: None


INFO:tensorflow:tokens: [CLS] subject : the best in extreme get access to the biggest movie archive on the net http : / / c ##g . seo ##mori ##a . com / aj ##y / i ##um . html remove yourself here [SEP]


INFO:tensorflow:tokens: [CLS] subject : the best in extreme get access to the biggest movie archive on the net http : / / c ##g . seo ##mori ##a . com / aj ##y / i ##um . html remove yourself here [SEP]


INFO:tensorflow:input_ids: 101 3395 1024 1996 2190 1999 6034 2131 3229 2000 1996 5221 3185 8756 2006 1996 5658 8299 1024 1013 1013 1039 2290 1012 27457 24610 2050 1012 4012 1013 19128 2100 1013 1045 2819 1012 16129 6366 4426 2182 102 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:input_ids: 101 3395 1024 1996 2190 1999 6034 2131 3229 2000 1996 5221 3185 8756 2006 1996 5658 8299 1024 1013 1013 1039 2290 1012 27457 24610 2050 1012 4012 1013 19128 2100 1013 1045 2819 1012 16129 6366 4426 2182 102 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:label: 1 (id = 1)


INFO:tensorflow:label: 1 (id = 1)


### Modeling

In [ ]:
# Compute train and warmup steps from batch size
# These hyperparameters are copied from this colab notebook (https://colab.sandbox.google.com/github/tensorflow/tpu/blob/master/tools/colab/bert_finetuning_with_cloud_tpus.ipynb)
BATCH_SIZE = 32
LEARNING_RATE = 1e-5
NUM_TRAIN_EPOCHS = 3.0
# Warmup is a period of time where hte learning rate 
# is small and gradually increases--usually helps training.
WARMUP_PROPORTION = 0.1
# Model configs
SAVE_CHECKPOINTS_STEPS = 500
SAVE_SUMMARY_STEPS = 100

# Compute # train and warmup steps from batch size
num_train_steps = int(len(train_features) / BATCH_SIZE * NUM_TRAIN_EPOCHS)
num_warmup_steps = int(num_train_steps * WARMUP_PROPORTION)

In [ ]:
# Specify outpit directory and number of checkpoint steps to save

OUTPUT_DIR = '/content/drive/My Drive/Bert testing/Output'

run_config = tf.estimator.RunConfig(
    model_dir=OUTPUT_DIR,
    save_summary_steps=SAVE_SUMMARY_STEPS,
    save_checkpoints_steps=SAVE_CHECKPOINTS_STEPS)

#### Build model function

In [ ]:
# Download the pre-trained model, telling what to use from it, and add additional calculation on the output from the pre-trained model
# here uses tf.matmul (vector multiplication) on top of the pretrained model output
# next convert this the result to logit

def create_model(is_predicting, input_ids, input_mask, segment_ids, labels,
                 num_labels):
  """Creates a classification model."""

  bert_module = hub.Module(
      BERT_MODEL_HUB,    
      trainable=True)
  bert_inputs = dict(
      input_ids=input_ids,
      input_mask=input_mask,
      segment_ids=segment_ids)
  bert_outputs = bert_module(
      inputs=bert_inputs,
      signature="tokens",
      as_dict=True)

  # Use "pooled_output" for classification tasks on an entire sentence.
  # Use "sequence_outputs" for token-level output.
  output_layer = bert_outputs["pooled_output"]

  hidden_size = output_layer.shape[-1].value

  # Create our own layer to tune for politeness data.
  output_weights = tf.get_variable(
      "output_weights", [num_labels, hidden_size],
      initializer=tf.truncated_normal_initializer(stddev=0.02))

  output_bias = tf.get_variable(
      "output_bias", [num_labels], initializer=tf.zeros_initializer())

  with tf.variable_scope("loss"):

    # Dropout helps prevent overfitting
    output_layer = tf.nn.dropout(output_layer, keep_prob=0.9)

    logits = tf.matmul(output_layer, output_weights, transpose_b=True)
    logits = tf.nn.bias_add(logits, output_bias)
    log_probs = tf.nn.log_softmax(logits, axis=-1)

    # Convert labels into one-hot encoding
    one_hot_labels = tf.one_hot(labels, depth=num_labels, dtype=tf.float32)

    predicted_labels = tf.squeeze(tf.argmax(log_probs, axis=-1, output_type=tf.int32))
    # If we're predicting, we want predicted labels and the probabiltiies.
    if is_predicting:
      return (predicted_labels, log_probs)

    # If we're train/eval, compute loss between predicted and actual label
    per_example_loss = -tf.reduce_sum(one_hot_labels * log_probs, axis=-1)
    loss = tf.reduce_mean(per_example_loss)
    return (loss, predicted_labels, log_probs)



In [ ]:
# code below creates an function to tell how and when to apply our model (above) for doing different tasks. 
# e.g. for training, for prediction
# if not used for prediction, then calculate evaluation metrics for it, e.g. f1_score, auc, recall etc.

# model_fn_builder actually creates our model function
# using the passed parameters for num_labels, learning_rate, etc.
def model_fn_builder(num_labels, learning_rate, num_train_steps,
                     num_warmup_steps):
  """Returns `model_fn` closure for TPUEstimator."""
  def model_fn(features, labels, mode, params):  # pylint: disable=unused-argument
    """The `model_fn` for TPUEstimator."""

    input_ids = features["input_ids"]
    input_mask = features["input_mask"]
    segment_ids = features["segment_ids"]
    label_ids = features["label_ids"]

    is_predicting = (mode == tf.estimator.ModeKeys.PREDICT)
    
    # TRAIN and EVAL
    if not is_predicting:

      (loss, predicted_labels, log_probs) = create_model(
        is_predicting, input_ids, input_mask, segment_ids, label_ids, num_labels)

      train_op = bert.optimization.create_optimizer(
          loss, learning_rate, num_train_steps, num_warmup_steps, use_tpu=False)

      # Calculate evaluation metrics. 
      def metric_fn(label_ids, predicted_labels):
        accuracy = tf.metrics.accuracy(label_ids, predicted_labels)
        f1_score = tf.contrib.metrics.f1_score(
            label_ids,
            predicted_labels)
        auc = tf.metrics.auc(
            label_ids,
            predicted_labels)
        recall = tf.metrics.recall(
            label_ids,
            predicted_labels)
        precision = tf.metrics.precision(
            label_ids,
            predicted_labels) 
        true_pos = tf.metrics.true_positives(
            label_ids,
            predicted_labels)
        true_neg = tf.metrics.true_negatives(
            label_ids,
            predicted_labels)   
        false_pos = tf.metrics.false_positives(
            label_ids,
            predicted_labels)  
        false_neg = tf.metrics.false_negatives(
            label_ids,
            predicted_labels)
        return {
            "eval_accuracy": accuracy,
            "f1_score": f1_score,
            "auc": auc,
            "precision": precision,
            "recall": recall,
            "true_positives": true_pos,
            "true_negatives": true_neg,
            "false_positives": false_pos,
            "false_negatives": false_neg
        }

      eval_metrics = metric_fn(label_ids, predicted_labels)

      if mode == tf.estimator.ModeKeys.TRAIN:
        return tf.estimator.EstimatorSpec(mode=mode,
          loss=loss,
          train_op=train_op)
      else:
          return tf.estimator.EstimatorSpec(mode=mode,
            loss=loss,
            eval_metric_ops=eval_metrics)
    else:
      (predicted_labels, log_probs) = create_model(
        is_predicting, input_ids, input_mask, segment_ids, label_ids, num_labels)

      predictions = {
          'probabilities': log_probs,
          'labels': predicted_labels
      }
      return tf.estimator.EstimatorSpec(mode, predictions=predictions)

  # Return the actual model function in the closure
  return model_fn

#### Model set up

In [ ]:
model_fn = model_fn_builder(
  num_labels=len(label_list),
  learning_rate=LEARNING_RATE,
  num_train_steps=num_train_steps,
  num_warmup_steps=num_warmup_steps)

estimator = tf.estimator.Estimator(
  model_fn=model_fn,
  config=run_config,
  params={"batch_size": BATCH_SIZE})

INFO:tensorflow:Using config: {'_model_dir': '/content/drive/My Drive/Bert testing/Output', '_tf_random_seed': None, '_save_summary_steps': 100, '_save_checkpoints_steps': 500, '_save_checkpoints_secs': None, '_session_config': allow_soft_placement: true
graph_options {
  rewrite_options {
    meta_optimizer_iterations: ONE
  }
}
, '_keep_checkpoint_max': 5, '_keep_checkpoint_every_n_hours': 10000, '_log_step_count_steps': 100, '_train_distribute': None, '_device_fn': None, '_protocol': None, '_eval_distribute': None, '_experimental_distribute': None, '_experimental_max_worker_delay_secs': None, '_session_creation_timeout_secs': 7200, '_service': None, '_cluster_spec': <tensorflow.python.training.server_lib.ClusterSpec object at 0x7f79b7e4a748>, '_task_type': 'worker', '_task_id': 0, '_global_id_in_cluster': 0, '_master': '', '_evaluation_master': '', '_is_chief': True, '_num_ps_replicas': 0, '_num_worker_replicas': 1}


INFO:tensorflow:Using config: {'_model_dir': '/content/drive/My Drive/Bert testing/Output', '_tf_random_seed': None, '_save_summary_steps': 100, '_save_checkpoints_steps': 500, '_save_checkpoints_secs': None, '_session_config': allow_soft_placement: true
graph_options {
  rewrite_options {
    meta_optimizer_iterations: ONE
  }
}
, '_keep_checkpoint_max': 5, '_keep_checkpoint_every_n_hours': 10000, '_log_step_count_steps': 100, '_train_distribute': None, '_device_fn': None, '_protocol': None, '_eval_distribute': None, '_experimental_distribute': None, '_experimental_max_worker_delay_secs': None, '_session_creation_timeout_secs': 7200, '_service': None, '_cluster_spec': <tensorflow.python.training.server_lib.ClusterSpec object at 0x7f79b7e4a748>, '_task_type': 'worker', '_task_id': 0, '_global_id_in_cluster': 0, '_master': '', '_evaluation_master': '', '_is_chief': True, '_num_ps_replicas': 0, '_num_worker_replicas': 1}


In [ ]:
# Create an input function for training.
train_input_fn = bert.run_classifier.input_fn_builder(
    features=train_features,
    seq_length=MAX_SEQ_LENGTH,
    is_training=True,
    drop_remainder=False) # drop_remainder = True for using TPUs.


In [ ]:
# training
print(f'Beginning Training!')
current_time = datetime.now()

estimator.train(input_fn=train_input_fn, max_steps=num_train_steps)
print("Training took time ", datetime.now() - current_time)

Beginning Training!
Instructions for updating:
Use Variable.read_value. Variables in 2.X are initialized automatically both in eager and graph (inside tf.defun) contexts.


Instructions for updating:
Use Variable.read_value. Variables in 2.X are initialized automatically both in eager and graph (inside tf.defun) contexts.


INFO:tensorflow:Calling model_fn.


INFO:tensorflow:Calling model_fn.


INFO:tensorflow:Saver not created because there are no variables in the graph to restore


INFO:tensorflow:Saver not created because there are no variables in the graph to restore
/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/framework/indexed_slices.py:424: UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a large amount of memory.
  "Converting sparse IndexedSlices to a dense Tensor of unknown shape. "


INFO:tensorflow:Done calling model_fn.


INFO:tensorflow:Done calling model_fn.


INFO:tensorflow:Create CheckpointSaverHook.


INFO:tensorflow:Create CheckpointSaverHook.


INFO:tensorflow:Graph was finalized.


INFO:tensorflow:Graph was finalized.


INFO:tensorflow:Running local_init_op.


INFO:tensorflow:Running local_init_op.


INFO:tensorflow:Done running local_init_op.


INFO:tensorflow:Done running local_init_op.


INFO:tensorflow:Saving checkpoints for 0 into /content/drive/My Drive/Bert testing/Output/model.ckpt.


INFO:tensorflow:Saving checkpoints for 0 into /content/drive/My Drive/Bert testing/Output/model.ckpt.


INFO:tensorflow:loss = 0.69442594, step = 0


INFO:tensorflow:loss = 0.69442594, step = 0


INFO:tensorflow:global_step/sec: 1.59523


INFO:tensorflow:global_step/sec: 1.59523


INFO:tensorflow:loss = 0.15728918, step = 100 (62.691 sec)


INFO:tensorflow:loss = 0.15728918, step = 100 (62.691 sec)


INFO:tensorflow:global_step/sec: 2.37894


INFO:tensorflow:global_step/sec: 2.37894


INFO:tensorflow:loss = 0.0029668945, step = 200 (42.035 sec)


INFO:tensorflow:loss = 0.0029668945, step = 200 (42.035 sec)


INFO:tensorflow:global_step/sec: 2.37987


INFO:tensorflow:global_step/sec: 2.37987


INFO:tensorflow:loss = 0.00063895347, step = 300 (42.019 sec)


INFO:tensorflow:loss = 0.00063895347, step = 300 (42.019 sec)


INFO:tensorflow:global_step/sec: 2.37812


INFO:tensorflow:global_step/sec: 2.37812


INFO:tensorflow:loss = 0.00051306153, step = 400 (42.050 sec)


INFO:tensorflow:loss = 0.00051306153, step = 400 (42.050 sec)


INFO:tensorflow:Saving checkpoints for 449 into /content/drive/My Drive/Bert testing/Output/model.ckpt.


INFO:tensorflow:Saving checkpoints for 449 into /content/drive/My Drive/Bert testing/Output/model.ckpt.


INFO:tensorflow:Loss for final step: 0.00045871772.


INFO:tensorflow:Loss for final step: 0.00045871772.


Training took time  0:04:26.654187


### Model evaluation

In [ ]:
test_input_fn = run_classifier.input_fn_builder(
    features=test_features,
    seq_length=max_length,
    is_training=False,
    drop_remainder=False)

In [ ]:
estimator.evaluate(input_fn=test_input_fn, steps=None)

INFO:tensorflow:Calling model_fn.


INFO:tensorflow:Calling model_fn.


INFO:tensorflow:Saver not created because there are no variables in the graph to restore


INFO:tensorflow:Saver not created because there are no variables in the graph to restore
/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/framework/indexed_slices.py:424: UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a large amount of memory.
  "Converting sparse IndexedSlices to a dense Tensor of unknown shape. "


INFO:tensorflow:Done calling model_fn.


INFO:tensorflow:Done calling model_fn.


INFO:tensorflow:Starting evaluation at 2020-09-05T03:39:19Z


INFO:tensorflow:Starting evaluation at 2020-09-05T03:39:19Z


INFO:tensorflow:Graph was finalized.


INFO:tensorflow:Graph was finalized.


INFO:tensorflow:Restoring parameters from /content/drive/My Drive/Bert testing/Output/model.ckpt-449


INFO:tensorflow:Restoring parameters from /content/drive/My Drive/Bert testing/Output/model.ckpt-449


INFO:tensorflow:Running local_init_op.


INFO:tensorflow:Running local_init_op.


INFO:tensorflow:Done running local_init_op.


INFO:tensorflow:Done running local_init_op.


INFO:tensorflow:Finished evaluation at 2020-09-05-03:39:33


INFO:tensorflow:Finished evaluation at 2020-09-05-03:39:33


INFO:tensorflow:Saving dict for global step 449: auc = 0.9946439, eval_accuracy = 0.99666667, f1_score = 0.9977477, false_negatives = 1.0, false_positives = 3.0, global_step = 449, loss = 0.011223778, precision = 0.9966254, recall = 0.9988726, true_negatives = 310.0, true_positives = 886.0


INFO:tensorflow:Saving dict for global step 449: auc = 0.9946439, eval_accuracy = 0.99666667, f1_score = 0.9977477, false_negatives = 1.0, false_positives = 3.0, global_step = 449, loss = 0.011223778, precision = 0.9966254, recall = 0.9988726, true_negatives = 310.0, true_positives = 886.0


INFO:tensorflow:Saving 'checkpoint_path' summary for global step 449: /content/drive/My Drive/Bert testing/Output/model.ckpt-449


INFO:tensorflow:Saving 'checkpoint_path' summary for global step 449: /content/drive/My Drive/Bert testing/Output/model.ckpt-449


{'auc': 0.9946439,
 'eval_accuracy': 0.99666667,
 'f1_score': 0.9977477,
 'false_negatives': 1.0,
 'false_positives': 3.0,
 'global_step': 449,
 'loss': 0.011223778,
 'precision': 0.9966254,
 'recall': 0.9988726,
 'true_negatives': 310.0,
 'true_positives': 886.0}

Evaluation over test data achieves astonishing accuracy score 99.67%. 

### Predictions

Testing model performances using newly made-up messages.

In [ ]:
def getPrediction(in_sentences):
  labels = ["Negative", "Positive"]
  input_examples = [run_classifier.InputExample(guid="", text_a = x, text_b = None, label = 0) for x in in_sentences] # here, "" is just a dummy label
  input_features = run_classifier.convert_examples_to_features(input_examples, label_list, MAX_SEQ_LENGTH, tokenizer)
  predict_input_fn = run_classifier.input_fn_builder(features=input_features, seq_length=MAX_SEQ_LENGTH, is_training=False, drop_remainder=False)
  predictions = estimator.predict(predict_input_fn)
  return [(sentence, prediction['probabilities'], labels[prediction['labels']]) for sentence, prediction in zip(in_sentences, predictions)]

In [ ]:
pred_sentences = [
  "Subject: you have won 10 million dollars, please email us your bank account and $500 deposits as soon as possible.",
  "Subject: confirming our doctor appointment tomorrow, please reply to confirm."
]

In [ ]:
predictions = getPrediction(pred_sentences)

INFO:tensorflow:Writing example 0 of 2


INFO:tensorflow:Writing example 0 of 2


INFO:tensorflow:*** Example ***


INFO:tensorflow:*** Example ***


INFO:tensorflow:guid: 


INFO:tensorflow:guid: 


INFO:tensorflow:tokens: [CLS] subject : you have won 10 million dollars , please email us your bank account and $ 500 deposits as soon as possible . [SEP]


INFO:tensorflow:tokens: [CLS] subject : you have won 10 million dollars , please email us your bank account and $ 500 deposits as soon as possible . [SEP]


INFO:tensorflow:input_ids: 101 3395 1024 2017 2031 2180 2184 2454 6363 1010 3531 10373 2149 2115 2924 4070 1998 1002 3156 10042 2004 2574 2004 2825 1012 102 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:input_ids: 101 3395 1024 2017 2031 2180 2184 2454 6363 1010 3531 10373 2149 2115 2924 4070 1998 1002 3156 10042 2004 2574 2004 2825 1012 102 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:label: 0 (id = 0)


INFO:tensorflow:label: 0 (id = 0)


INFO:tensorflow:*** Example ***


INFO:tensorflow:*** Example ***


INFO:tensorflow:guid: 


INFO:tensorflow:guid: 


INFO:tensorflow:tokens: [CLS] subject : confirming our doctor appointment tomorrow , please reply to confirm . [SEP]


INFO:tensorflow:tokens: [CLS] subject : confirming our doctor appointment tomorrow , please reply to confirm . [SEP]


INFO:tensorflow:input_ids: 101 3395 1024 19195 2256 3460 6098 4826 1010 3531 7514 2000 12210 1012 102 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:input_ids: 101 3395 1024 19195 2256 3460 6098 4826 1010 3531 7514 2000 12210 1012 102 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:label: 0 (id = 0)


INFO:tensorflow:label: 0 (id = 0)


INFO:tensorflow:Calling model_fn.


INFO:tensorflow:Calling model_fn.


INFO:tensorflow:Saver not created because there are no variables in the graph to restore


INFO:tensorflow:Saver not created because there are no variables in the graph to restore


INFO:tensorflow:Done calling model_fn.


INFO:tensorflow:Done calling model_fn.


INFO:tensorflow:Graph was finalized.


INFO:tensorflow:Graph was finalized.


INFO:tensorflow:Restoring parameters from /content/drive/My Drive/Bert testing/Output/model.ckpt-449


INFO:tensorflow:Restoring parameters from /content/drive/My Drive/Bert testing/Output/model.ckpt-449


INFO:tensorflow:Running local_init_op.


INFO:tensorflow:Running local_init_op.


INFO:tensorflow:Done running local_init_op.


INFO:tensorflow:Done running local_init_op.


In [ ]:
predictions

[('Subject: you have won 10 million dollars, please email us your bank account and $500 deposits as soon as possible.',
  array([-5.9292703e+00, -2.6639947e-03], dtype=float32),
  'Positive'),
 ('Subject: confirming our doctor appointment tomorrow, please reply to confirm.',
  array([-0.05193083, -2.9836953 ], dtype=float32),
  'Negative')]

The prediction is exactly correct, where my first message is a typical fraud message I often see in my junk box and I usually receive the second type of message for appointments.